subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalInputsIntegrationTest.groovy

```diff
@@ -277,23 +277,36 @@ class IncrementalInputsIntegrationTest extends AbstractIncrementalTasksIntegrati
                 @TaskAction
                 void run(InputChanges changes) {
                     new File(outputDirectory, "output.txt").text = "Success"
+                    changes.getFileChanges(input).each {
+                        println it
+                    }                    
                 }
             }
 
             task myTask(type: MyTask) {
                 outputDirectory = file("build/output")
+                if (project.findProperty("inputFile")) {
+                    input = file(project.property("inputFile"))
+                }
             }
         """
 
         file("input").createDir()
 
         expect:
-        fails("myTask")
-        failure.assertHasDescription("Execution failed for task ':myTask'.")
-        failure.assertHasCause("Must specify a value for incremental input property 'input'.")
+        succeeds("myTask")
+
+        when:
+        file("inputFile").createDir()
+        file("inputFile/someContents.txt").text = "input"
+        run("myTask", "-PinputFile=inputFile")
+
+        then:
+        output.contains("Input file ${file("inputFile").absolutePath} has been added.")
+        output.contains("Input file ${file("inputFile/someContents.txt").absolutePath} has been added.")
```

Comment:
I think it would be good to prefix any custom output we are trying to match in an integration test with something like `> ` so that it's easier to figure out where it is printed from.

Reply:
Done.